core/jobs/types/model_property.py

```diff
@@ -109,62 +128,93 @@ def yield_value_from_model(self, model):
         """Returns the model class associated with this instance.
 
         Returns:
             type(BaseModel). The model type.
         """
-        return job_utils.get_model_class(self._model_kind)
-
-    def _to_property(self):
+        # Here, get_model_class() returns the value of type
+        # Type[datastore_services.Model] but the expected return value from
+        # this method is Type[base_models.BaseModel]. So, that's why we used
+        # cast here.
+        return cast(
+            Type[base_models.BaseModel],
+            job_utils.get_model_class(self._model_kind)
+        )
+
+    def _to_property(
+        self
+    ) -> Union[
+        datastore_services.Property,
+        Callable[[base_models.BaseModel], str]
+    ]:
         """Returns the Property object associated with this instance.
 
         Returns:
             *. A property instance.
         """
-        return getattr(self._to_model_class(), self._property_name)
-
-    def _is_repeated_property(self):
+        # Using a cast here because the return value of getattr() method is
+        # dynamic and mypy will assume it to be Any type otherwise.
+        return cast(
+            Union[
+                datastore_services.Property,
+                Callable[[base_models.BaseModel], str]
+            ],
+            getattr(self._to_model_class(), self._property_name)
+        )
+
+    def _is_repeated_property(self) -> bool:
         """Returns whether the property is repeated.
 
         Returns:
             bool. Whether the property is repeated.
         """
-        return self._property_name != 'id' and self._to_property()._repeated # pylint: disable=protected-access
+        model_property = self._to_property()
+        if (
+            self._property_name != 'id' and
+            isinstance(model_property, datastore_services.Property)
+        ):
+            return model_property._repeated  # pylint: disable=protected-access
+        else:
+            return False
 
-    def __getstate__(self):
+    def __getstate__(self) -> Tuple[str, str]:
         """Called by pickle to get the value that uniquely defines self.
 
         Returns:
             tuple(str, str). The model's kind and the name of the property.
         """
         return self._model_kind, self._property_name
 
-    def __setstate__(self, state):
+    def __setstate__(self, state: Tuple[str, str]) -> None:
         """Called by pickle to build an instance from __getstate__'s value.
 
         Args:
             state: tuple(str, str). The model's kind and the property's name.
         """
         self._model_kind, self._property_name = state
 
-    def __str__(self):
+    def __str__(self) -> str:
         return '%s.%s' % (self._model_kind, self._property_name)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return 'ModelProperty(%s, %s)' % (self._model_kind, self)
 
-    def __eq__(self, other):
+    # NOTE: Needs to return Any because of:
```

Comment:
```suggestion
    # NOTE: Needs to return Any because the function could also return NotImplemented:
```

Reply:
A similar comment will give a gist of what the issue is.

Reply:
Done!